pkg/operator/nodekubeconfigcontroller/nodekubeconfigcontroller.go

```diff
@@ -0,0 +1,181 @@
+		kubeInformersForNamespaces.InformersFor(operatorclient.OperatorNamespace).Core().V1().ConfigMaps().Informer(),
+		kubeInformersForNamespaces.InformersFor(operatorclient.TargetNamespace).Core().V1().ConfigMaps().Informer(),
+		kubeInformersForNamespaces.InformersFor(operatorclient.OperatorNamespace).Core().V1().Secrets().Informer(),
+		kubeInformersForNamespaces.InformersFor(operatorclient.TargetNamespace).Core().V1().Secrets().Informer(),
+		infrastuctureInformer.Informer(),
+	).WithSync(c.sync).ResyncEvery(time.Second).ToController("NodeKubeconfigController", eventRecorder.WithComponentSuffix("node-kubeconfig-controller"))
+}
+
+func (c NodeKubeconfigController) sync(ctx context.Context, syncContext factory.SyncContext) error {
+	operatorSpec, _, _, err := c.operatorClient.GetStaticPodOperatorState()
+	if err != nil {
+		return err
+	}
+
+	switch operatorSpec.ManagementState {
+	case operatorv1.Managed:
+	case operatorv1.Unmanaged:
+		return nil
+	case operatorv1.Removed:
+		// TODO probably just fail
+		return nil
+	default:
+		syncContext.Recorder().Warningf("ManagementStateUnknown", "Unrecognized operator management state %q", operatorSpec.ManagementState)
+		return nil
+	}
+
+	var errors []error
+
+	err = ensureNodeKubeconfigs(
+		ctx,
+		c.kubeClient.CoreV1(),
+		c.secretLister,
+		c.configMapLister,
+		c.infrastuctureLister,
+		syncContext.Recorder(),
+	)
+	if err != nil {
+		errors = append(errors, fmt.Errorf("%q: %v", "secret/node-kubeconfigs", err))
+	}
+
+	if len(errors) > 0 {
+		condition := operatorv1.OperatorCondition{
+			Type:    "NodeKubeconfigController",
+			Status:  operatorv1.ConditionTrue,
+			Reason:  "SynchronizationError",
+			Message: v1helpers.NewMultiLineAggregate(errors).Error(),
+		}
+		_, _, err = v1helpers.UpdateStaticPodStatus(c.operatorClient, v1helpers.UpdateStaticPodConditionFn(condition))
+		if err != nil {
+			return err
+		}
+
+		syncContext.Queue().Add(workQueueKey)
+		return nil
+	}
+
+	condition := operatorv1.OperatorCondition{
+		Type:   "NodeKubeconfigController",
+		Status: operatorv1.ConditionFalse,
+	}
+	_, _, err = v1helpers.UpdateStaticPodStatus(c.operatorClient, v1helpers.UpdateStaticPodConditionFn(condition))
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func ensureNodeKubeconfigs(ctx context.Context, client coreclientv1.CoreV1Interface, secretLister corev1listers.SecretLister, configmapLister corev1listers.ConfigMapLister, infrastructureLister configv1listers.InfrastructureLister, recorder events.Recorder) error {
+	requiredSecret := resourceread.ReadSecretV1OrDie(v410_00_assets.MustAsset("v4.1.0/kube-apiserver/node-kubeconfigs.yaml"))
+
+	systemAdminCredsSecret, err := secretLister.Secrets(operatorclient.OperatorNamespace).Get("node-system-admin-client")
+	if err != nil {
+		return err
+	}
+
+	systemAdminClientCert := systemAdminCredsSecret.Data[corev1.TLSCertKey]
+	if len(systemAdminClientCert) == 0 {
+		return fmt.Errorf("missing system:admin client certificate")
```

Comment:
nit: Would it make sense to provide an indication of where this data is coming from (i.e. the secret name)? Without that detail someone seeing this error in the log needs to know where the data is coming from or refer to the code to find out vs the error message having enough detail e.g.

 - `secret/node-kubeconfigs: missing system:admin client certificate`

vs

 - `secret/node-kubeconfigs: system:admin client certificate missing from secret/node-system-admin-client`